samples/excel/90-scenarios/currency-converter.yaml

```diff
@@ -0,0 +1,178 @@
+author: cakriwut
+host: EXCEL
+api_set:
+    ExcelApi: '1.4'
+script:
+    content: |
+        /*
+        * Copyright (c) Riwut Libinuko. All rights reserved. Licensed under the MIT license.
+        */
+
+        declare let moment: any;
+        const tableName = "TransactionTable";
+
+        let tblSetup: Record<string, number> = {};
+
+        $("#convert").click(() => tryCatch(convert));
+        $("#setup").click(() => tryCatch(setupSample));
+
+        /** Main converting function **/
+        async function convert() {
+          await Excel.run(async (context) => {
+            const table = context.workbook.tables.getItem(tableName)
+            table.columns.load("items");
+            await context.sync();
+            if (!IsTableValid(table.columns.items)) {
+              console.error('Error: Some of default table header is missing. Required headers: \r\n"Currency","Price (Currency)","Transaction Date","Base","Price in Base"')
+              return;
+            }
+
+            table.rows.load("items,values");
+            await context.sync();
+            asyncForEach(table.rows.items, async (row,idx,arr) => {
+            const priceBaseRange = row.getRange().getCell(0, tblSetup['Price in Base']);
+
+            priceBaseRange.load("address,text");
+            context.trackedObjects.add(priceBaseRange);
+            await context.sync();
+            const priceInBase = row.values[0][tblSetup['Price in Base']];
+            const priceInCurrency = row.values[0][tblSetup['Price (Currency)']];
+            const transactionDate = row.values[0][tblSetup['Transaction Date']]
+            const currency = row.values[0][tblSetup['Currency']];
+            const baseCurrency = row.values[0][tblSetup['Base']]
+            
+            if (priceInBase === "") {
+              const dateMoment = moment.fromOADate(transactionDate);
+              const period = dateMoment.format("YYYY-MM-DD");
+              const queryUrl = convertValue(currency, baseCurrency, period, period);
+              const response = await fetch(queryUrl);
+              if (response.ok) {
+                const result = await response.json();
+                const converted = priceInCurrency * result.rates[baseCurrency];
+                priceBaseRange.values = [[converted]];
+                await context.sync();
+                context.trackedObjects.remove(priceBaseRange);
+              }
+            } else {
+                console.info(`Row: ${idx} has been converted. Skipped.`)
+              }
+            });
+          });
+        }
+
+        /** Async ForEeach */
+        async function asyncForEach(array: Array<any>, callback) {
+          for(let idx = 0; idx < array.length; idx++) {
+            await callback(array[idx],idx, array);
+          }
+        }
+
+        /** Check if Table is Valid, ie has all required column **/
+        function IsTableValid(items: Excel.TableColumn[]) {
```

Comment:
Thanks for clarifying the purpose of this method. Since you want to highlight the possibility of a dynamic table ordering, let's make that clear in the comment.
"Check if the table contains the necessary columns. These columns can be in any order within the table."

Reply:
Updated.